### PR TITLE
added trusted_account support

### LINKED ISSUE #574: Improvement: allow another aws account to assume the iam role that has write access to a kinesis stream
## Background

From: xinghong.fang

### Description

New feature suggest: “allow another aws account to assume the iam role that has write access to a kinesis stream”.

I think this will be useful as log collectors running on other aws account can assume the kinesis_writeonly role to write logs to it without having to provide iam user credential. I have some rough idea on how to implement this, let me know if you are interested

it looks roughly like this in the cluster config file:

```
    "kinesis": {
      "streams": {
        "retention": 24,
        "shards": 1,
        "trusted_account": "12345678"
      }
    },
```

the default will be “”, and terraform will only create the role if the value is non-empty

### PR DESCRIPTION
**Pull request by javefang**
to: @airbnb/streamalert-maintainers
cc: <optional-cc-to-specific-users>
size: small
resolves #574 

## Background

Add new feature to allow kinesis write permission to be granted to other AWS accounts.

## Changes

* Added kinesis_stream option "trusted_accounts" (list of strings)
* Added Terraform scripts to generate new IAM role which grants kinesis write permission to the "trusted_accounts".

## Testing

Steps for how this change was tested and verified

```
23:43 $ tests/scripts/pylint.sh
Starting pylint script

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

```
00:14 $ ./tests/scripts/unit_tests.sh
...
CLI - Terraform Generate Kinesis Streams ... ok (0.0004s)
...
TOTAL                                                    3047    119    96%
[success] 9.88% tests.unit.stream_alert_rule_processor.test_threat_intel.TestStreamThreatIntel.test_query_with_exception: 1.3112s
[success] 3.60% tests.unit.stream_alert_rule_processor.test_threat_intel.TestStreamThreatIntel.test_process_ioc_with_clienterror: 0.4773s
[success] 3.38% tests.unit.stream_alert_athena_partition_refresh.test_main.TestStreamAlertAthenaGlobals.test_handler_no_unique_buckets: 0.4486s
[success] 1.43% tests.unit.stream_alert_alert_processor.test_outputs.test_pagerduty.TestPagerDutyIncidentOutput.test_dispatch_failure_bad_everything: 0.1895s
[success] 1.38% tests.unit.stream_alert_alert_processor.test_outputs.test_jira.TestJiraOutput.test_dispatch_issue_new: 0.1833s
[success] 1.35% tests.unit.stream_alert_alert_processor.test_outputs.test_aws.TestS3Ouput.test_dispatch: 0.1786s
[success] 1.31% tests.unit.app_integrations.test_apps.test_app_base.TestAppIntegration.test_check_http_response_bad: 0.1742s
[success] 1.28% tests.unit.stream_alert_alert_processor.test_outputs.test_phantom.TestPhantomOutput.test_dispatch_check_container_no_response: 0.1701s
[success] 1.22% tests.unit.stream_alert_alert_processor.test_outputs.test_jira.TestJiraOutput.test_auth_empty_response: 0.1619s
[success] 1.08% tests.unit.stream_alert_cli.test_config.TestCLIConfig.test_add_threat_intel_with_table_name: 0.1431s
----------------------------------------------------------------------
Ran 518 tests in 13.459s

OK
```

### CODE REVIEW

**Comment by javefang** [comment]
Need an LGTM :) @jacknagz

**Comment by javefang** [comment]
Missed the pylint, apologise for that. I ran it locally first this time, should be fixed now.

**Comment by jacknagz** [comment]
Thanks @javefang!! More quick questions for thought on this change: Should we allow for multiple trusted accounts? And can this just default to the account ID specified in the global config?

File: tests/unit/stream_alert_cli/terraform/test_kinesis_streams.py [hunk 1]
```diff
@@ -39,7 +39,8 @@ def test_kinesis_streams():
                 'stream_name': 'unit-testing_advanced_stream_alert_kinesis',
                 'shards': 1,
                 'retention': 24,
-                'create_user': True
+                'create_user': True,
+                'trusted_account': ''
```
[the remark this answers is not in the document]
Comment (javefang) [reply tests/unit/stream_alert_cli/terraform/test_kinesis_streams.py]: test case added

**Comment by javefang** [comment]
@jacknagz 

> Should we allow for multiple trusted accounts? 
I would stick with a single account for now for simplicity. Although if we think there is a good chance we might have to add this later then it's probably better to do it now to avoid it becoming a breaking change.

> And can this just default to the account ID specified in the global config?
If it is the ec2 instance in the same account assuming the role. How should we define the `aws_iam_policy_document`? I feel like there must have a way without having to mention the account id of the current account? Do you have an example terraform script for to implement this IAM config?

**Comment by jacknagz** [comment]
@javefang cool, I think we can just accept a list of account IDs and we should be good.  On the Terraform side, you can use [formatlist](https://www.terraform.io/docs/configuration/interpolation.html#formatlist-format-args-) to generate the policy for multiple accounts

**Comment by javefang** [comment]
@jacknagz multi account support implemented, also rebased from the latest master head.

**Comment by jacknagz** [comment]
Thanks @javefang !!!

**Pull request merged**